Fix a panic in Route's new fee-calculation methods and clean up

TheBlueMatt opened the pull request:
This addresses Val's feedback on the new Route fee- and
amount-calculation methods, including fixing the panic she
identified and cleaning up various docs and comments.

See https://github.com/rust-bitcoin/rust-lightning/pull/1063#pullrequestreview-760035794

jkczyz commented on lightning/src/routing/router.rs:
```diff
@@ -72,21 +72,22 @@ pub struct Route {
 		// Do not count last hop of each path since that's the full value of the payment
 		return self.paths.iter()
-			.flat_map(|path| path.split_last().unwrap().1)
+			.flat_map(|path| path.split_last().map(|path_pfx| path_pfx.1).unwrap_or(&[]))
```
This may be more readable if you destructure the tuple as `(_, path_prefix)` given only the last part of the tuple is the prefix, assuming `pfx` stands for prefix. Wasn't sure.

jkczyz commented on lightning/src/routing/router.rs:
```diff
@@ -72,21 +72,22 @@ pub struct Route {
+	/// Returns the total amount of fees paid on this [`Route`].
+	///
+	/// This doesn't include any extra payment made to the recipient, which can happen in excess of
+	/// the amount passed to [``get_route``]'s `final_value_msat`.
```
Remove extra ticks.

jkczyz commented on lightning/src/routing/router.rs:
```diff
@@ -72,21 +72,22 @@ pub struct Route {
 		return self.paths.iter()
-			.map(|path| path.split_last().unwrap().0)
-			.map(|hop| &hop.fee_msat)
+			.map(|path| path.split_last().map(|hop| hop.0.fee_msat).unwrap_or(0))
```
Similar.

TheBlueMatt commented:
Addressed some comments from Jeff and squashed, will land after CI, diff is:

```
$ git diff-tree -U1 6ccbe887 59925c1f
diff --git a/lightning/src/routing/router.rs b/lightning/src/routing/router.rs
index af783954..e60fd1ee 100644
--- a/lightning/src/routing/router.rs
+++ b/lightning/src/routing/router.rs
@@ -77,3 +77,3 @@ impl Route {
 	/// This doesn't include any extra payment made to the recipient, which can happen in excess of
-	/// the amount passed to [``get_route``]'s `final_value_msat`.
+	/// the amount passed to [`get_route`]'s `final_value_msat`.
 	pub fn get_total_fees(&self) -> u64 {
@@ -81,3 +81,3 @@ impl Route {
 		return self.paths.iter()
-			.flat_map(|path| path.split_last().map(|path_pfx| path_pfx.1).unwrap_or(&[]))
+			.flat_map(|path| path.split_last().map(|(_, path_prefix)| path_prefix).unwrap_or(&[]))
 			.map(|hop| &hop.fee_msat)
@@ -89,3 +89,3 @@ impl Route {
 		return self.paths.iter()
-			.map(|path| path.split_last().map(|hop| hop.0.fee_msat).unwrap_or(0))
+			.map(|path| path.split_last().map(|(hop, _)| hop.fee_msat).unwrap_or(0))
 			.sum();
$
```
